Bottom: 09bc532609a2f4c6d9423a71ee9e45ee20e52068
Top:    1111bd164f8b241d253eacb6f67b970b98db2215
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-16 13:23:03 -0500

Refresh of add-buildbot-ptxdist-worker

---
diff --git a/roles/buildbot-worker-ptxdist/defaults/main.yml b/roles/buildbot-worker-ptxdist/defaults/main.yml
index 7fb5f918..11c44750 100644
--- a/roles/buildbot-worker-ptxdist/defaults/main.yml
+++ b/roles/buildbot-worker-ptxdist/defaults/main.yml
@@ -39,6 +39,12 @@ container_hostname: '{{ role_name }}'
 # the name of the container being started
 container_name: '{{ docker_name_prefix }}{{ container_hostname }}'
 
+# network configuration for the confainer
+container_networks: []
+
+# port configuration of the container
+container_port_args: []
+
 # volumes mounted within the container
 container_volumes:
 - '{{ data_volume }}:{{ data_path }}:z'
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
index 1a45e06a..9b5be868 100644
--- a/roles/buildbot-worker-ptxdist/tasks/main.yml
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -46,18 +46,15 @@
     cleanup: yes
     command: /bin/ash -c 
       "mkdir -p {{item}} &&
-      chown -c -R {{buildbot_uid}}:{{buildbot_uid}} {{item}}"
+      if [ $(stat --printf='%u' '{{item}}') -ne {{buildbot_uid}} ]; then
+        chown -c -R {{buildbot_uid}}:{{buildbot_uid}} {{item}};
+      fi"
 
 - name: Create secrets volume
   file:
     path: '{{ secrets_hostdir }}'
     state: directory
 
-- name: Populate known_hosts secret
-  copy:
-    content: '{{ known_hosts }}'
-    dest: '{{ secrets_hostdir }}/{{ secrets_known_hosts }}'
-
 - name: Checkout image repo
   git:
     repo: '{{ image_repo }}'
@@ -72,14 +69,15 @@
     buildargs: '{{ image_args }}'
     force: true
 
-- name: Start buildbot container
+- name: Start buildbot worker container
   docker_container:
+    hostname: '{{ container_hostname }}'
     name: '{{ container_name }}'
     image: '{{ image_name }}:{{ docker_image_tag }}'
-    # networks: '{{ buildbot_worker_ptxdist_networks }}'
+    networks: '{{ container_networks }}'
     volumes: '{{ container_volumes }}'
     env: '{{ container_env }}'
-    # ports: '{{ buildbot_worker_ptxdist_port_args }}'
+    ports: '{{ container_port_args }}'
 
 - name: Grab ssh public key
   command: >
@@ -87,7 +85,7 @@
   changed_when: false
   register: output
 
-- name: Add buildbot_worker_ptxdist host
+- name: Add buildbot worker host
   add_host:
-    name: buildbot_worker_ptxdist
+    name: "{{ container_hostname }}"
     pubkey: "{{ output.stdout }}"
